controller.go

```diff
@@ -686,19 +637,6 @@ func (c *controller) GetPluginGetter() plugingetter.PluginGetter {
 	return c.drvRegistry.GetPluginGetter()
 }
 
-func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver, capability driverapi.Capability) error {
-	c.Lock()
-	hd := c.discovery
-	c.Unlock()
-
-	if hd != nil {
-		c.pushNodeDiscovery(driver, capability, hd.Fetch(), true)
-	}
-
-	c.agentDriverNotify(driver)
```

Comment:
Shouldn't we keep this part? The lock and the block if `hd != nil` should be removed, but that line is independent from the node discovery I think.